File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/director_extend_runme.pl
use strict;
use warnings;
use Test::More tests => 5;
BEGIN { use_ok 'director_extend' }
require_ok 'director_extend';

{
  package MyObject;
  use base 'director_extend::SpObject';
  sub getFoo { 123 }
}

my $m = MyObject->new();
isa_ok $m, 'MyObject';
is($m->dummy(), 666, '1st call');
is($m->dummy(), 666, '2nd call');

File: support_tools/swigwin-3.0.12/Examples/test-suite/php5/director_extend_runme.php
<?php

require "tests.php";
require "director_extend.php";

// No new functions
check::functions(array(spobject_getfoobar,spobject_dummy,spobject_exceptionmethod));
// No new classes
check::classes(array(SpObject));
// now new vars
check::globals(array());

class MyObject extends SpObject{
  function getFoo() {
    return 123;
  }
}

$m = new MyObject();
check::equal($m->dummy(), 666, "1st call");
check::equal($m->dummy(), 666, "2st call"); // Locked system

check::done();
?>
